sentinel1_reader/sentinel1_reader.py

```diff
@@ -0,0 +1,331 @@
+
+    # calculate frequency for all slant range
+    freq_1d = np.array([doppler_poly1d.eval(t) for t in slant_ranges])
+
+    # init LUT2d (vstack freq since no az dependency) and return
+    return isce3.core.LUT2d(slant_ranges, az_times,
+                            np.vstack((freq_1d, freq_1d)))
+
+def get_burst_centers(tree):
+    '''
+    Parse grid points list and calculate burst center lat and lon
+
+    Parameters:
+    -----------
+    tree : Element
+        Element containing geolocation grid points.
+
+    Returns:
+    --------
+    _ : list
+        List of burst centers in degree longitude, latitude tuples.
+    _ : list
+        List of burst boundaries as list of degree longitude, latitude tuples.
+    '''
+    # find element tree
+    grid_pt_list = tree.find('geolocationGrid/geolocationGridPointList')
+
+    # read in all points
+    n_grid_pts = int(grid_pt_list.attrib['count'])
+    lines = np.empty(n_grid_pts)
+    pixels = np.empty(n_grid_pts)
+    lats = np.empty(n_grid_pts)
+    lons = np.empty(n_grid_pts)
+    for i, grid_pt in enumerate(grid_pt_list):
+        lines[i] = int(grid_pt[2].text)
+        pixels[i] = int(grid_pt[3].text)
+        lats[i] = float(grid_pt[4].text)
+        lons[i] = float(grid_pt[5].text)
+
+    unique_line_indices = np.unique(lines)
+    n_bursts = len(unique_line_indices) - 1
+    center_pts = [[]] * n_bursts
+    boundary_pts = [[]] * n_bursts
+
+    # zip lines numbers of bursts together and iterate
+    for i, (ln0, ln1) in enumerate(zip(unique_line_indices[:-1],
+                                       unique_line_indices[1:])):
+        # create masks for lines in current burst
+        mask0 = lines==ln0
+        mask1 = lines==ln1
+
+        # reverse order of 2nd set of points so plots of boundaries
+        # are not connected by a diagonal line
+        burst_lons = np.concatenate((lons[mask0], lons[mask1][::-1]))
+        burst_lats = np.concatenate((lats[mask0], lats[mask1][::-1]))
+
+        center_lon = np.mean(burst_lons)
+        center_lat = np.mean(burst_lats)
+        center_pts[i] = (center_lon, center_lat)
+
+        boundary_pts[i] = [(lon, lat) for lon, lat in zip(burst_lons,
+                                                          burst_lats)]
+
+    return center_pts, boundary_pts
+
+def xml2bursts(annotation_path: str, tiff_path: str, open_method=open):
+    '''
+    Parse bursts in Sentinel 1 annotation XML.
+
+    Parameters:
+    -----------
+    annotation_path : str
+        Path to Sentinel 1 annotation XML file.
+    tiff_path : str
+        Path to tiff file holding Sentinel 1 SLCs.
+    open_method : function
+        Function used to open annotation file.
+
+    Returns:
+    --------
+    bursts : list
+        List of Sentinel1BurstSlc objects found in annotation XML.
+    '''
+    _, tail = os.path.split(annotation_path)
+    platform_id, subswath_id, _, pol = [x.upper() for x in tail.split('-')[:4]]
+    burst_interval = 2.758277
```

Comment:
Can you write a comment on what we indicate as burst interval and how it is computed?

Reply:
The link below might be useful to add as well:
```
For IW mode, one burst has a duration of ~2.75 seconds and a burst overlap of approximately ~0.4 seconds.
https://sentinels.copernicus.eu/web/sentinel/user-guides/sentinel-1-sar/product-types-processing-levels/level-1
```